src/object/cloneDeep.ts

```diff
@@ -85,7 +85,10 @@ export function cloneDeep<T>(obj: T): Resolved<T> {
     return result as Resolved<T>;
   }
 
-  if (obj instanceof ArrayBuffer || obj instanceof SharedArrayBuffer) {
+  if (
+    obj instanceof ArrayBuffer ||
+    (typeof window.SharedArrayBuffer !== 'undefined' && obj instanceof SharedArrayBuffer)
```

Comment:
```suggestion
    (typeof SharedArrayBuffer !== 'undefined' && obj instanceof SharedArrayBuffer)
```

Since this library can be used in Node.js, we should not implicitly use window here.